python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -198,31 +217,62 @@ def execute(self, settings: CLISettings) -> None:
                 if self.as_hashes:
                     results = _match_hashes(path, s_type, index)
                 else:
-                    results = _match_file(path, s_type, index)
+                    results = _match_file(path, s_type, index, rotations=self.rotations)
 
                 for r in results:
                     metadatas: t.List[t.Tuple[str, FetchedSignalMetadata]] = r.metadata
                     for collab, fetched_data in metadatas:
                         if not self.all and collab in seen:
                             continue
                         seen.add(collab)
+
+                        # Add rotation information if possible 
+                        rotation_info = ""
+                        if hasattr(r.similarity_info, "rotation"):
+                            rotation_info = f" [{r.similarity_info.rotation.name}]"
+
                         # Supposed to be without whitespace, but let's make sure
                         distance_str = "".join(r.similarity_info.pretty_str().split())
                         print(
                             s_type.get_name(),
-                            distance_str,
+                            distance_str + rotation_info,
                             f"({collab})",
                             fetched_data,
                         )
 
 
 def _match_file(
-    path: pathlib.Path, s_type: t.Type[SignalType], index: SignalTypeIndex
+    path: pathlib.Path, 
+    s_type: t.Type[SignalType], 
+    index: SignalTypeIndex,
+    rotations: bool = False
 ) -> t.Sequence[IndexMatch]:
     if issubclass(s_type, MatchesStr):
         return index.query(path.read_text())
     assert issubclass(s_type, FileHasher)
-    return index.query(s_type.hash_from_file(path))
+
+    if not rotations or s_type != PhotoContent:
+        return index.query(s_type.hash_from_file(path))
+    
+    # Handle rotations for photos
+    with open(path, "rb") as f:
+        image_data = f.read()
+    
+    rotations = PhotoContent.all_simple_rotations(image_data)
+    all_matches = []
+
+    for rotation_type, rotated_bytes in rotations.items():
+        # Create a temporary BytesIO object to simulate a file 
+        temp_buffer = io.BytesIO(rotated_bytes)
+        matches = index.query(s_type.hash_from_file(temp_buffer))
+
+        # Add rotation information if any matches were found 
+        for match in matches: 
+            match.similarity_info.rotation = rotation_type
```

Comment:
blocking: I think this might be an illegal operation, and I'm unsure whether we should add rotation to the base concept of similarity. 

We have a few options, but the simplest one might be change the return of this function from IndexMatch to instead return a new dataclass that might contains the rotation information. 

I'll accept any solution that:
1. Doesn't change the base interface (since I don't think we yet know how to handle rotations in general - you are pioneering our approach!)
2. Can show which rotation was used in `tx match`.